app/controllers/course/survey/responses_controller.rb

```diff
@@ -77,7 +77,7 @@ def handle_create_error(error)
     @response = @survey.responses.accessible_by(current_ability).
                 find_by(course_user_id: current_course_user.id)
     if @response
-      render json: { responseId: @response.id }, status: :bad_request
+      render partial: 'created', status: :bad_request
```

Comment:
Technically, in the current request, the response was not created. I think it is better to use a 4xx code instead of a 2xx code since this is not 'normal' behaviour we expect, but an exception we want to recover from. What do you think?

Actually, from the description, 
> The request has been fulfilled, resulting in the creation of a new resource.

it seems appropriate for all `create` actions to respond with 201.